[Zhang zhong ming] iP

snowbanana12345 opened the pull request.

snowbanana12345 committed:
CAN YOU SEE THE CHANGE

snowbanana12345 committed:
Level-0 implemented

snowbanana12345 committed:
add dukeGreet function
add dukeGoodbye function
add duke echo user input functionality
add duke exiting if user input is bye

snowbanana12345 committed:
add static variables to store user inputs
add dukeRepeatInputs funcion which outputs stored user inputs

snowbanana12345 committed:
change duke response to reflect his deillusioned view of the world

add Task class :
a Task class is to model a user task
it can keep track of if the task is done
it can set the task to done

change storedUserInputs variable to store a Task array
add dukeSetDone function to enable a Task to be set to done

snowbanana12345 committed:
Wrote function to abstract store tasks in main function.
Add dukeStart and dukeStop function to abstract away setting a boolean variable to true or false
Fix a bug where duke cannot set the last added task to done.

snowbanana12345 committed:
Change task undone icon to cross
Change User Input System:
User will now first give one of the commands: bye, task, done, list
and then give the input

snowbanana12345 committed:
Abstract away all print out messages to duke level functions

snowbanana12345 committed:
Add DeadLine ToDo Event subclasses of Task
Task constructor no longer public
Replac done method in Task with isDone and setDone getter and setter
Add new commands: todo, deadline, event
Remov command: task
Add new methods: dukeAddToDo, dukeAddDeadLine, dukeAddEvent

snowbanana12345 committed:
slap project even more

snowbanana12345 committed:
add clearer instructions to command prompt
add dukeDefaultResponse method to ask if the user wants to see the list of command prompts

snowbanana12345 committed:
rename task event deadline todo into new task module
add duke_exception module
Methods dukeAddDeadLine dukeAddEvent dukeAddTodo now throws EmptyInputException if descriptions are not provided
add try catch block to handle new exceptions

snowbanana12345 committed:
Add enum class for duke commands
Add class dukeCommandManager to convert user input String to command Enum class
Method dukeCollectUserInput now throws InvalidCommandException if command is not valid. The
checking of the command is done in the dukeCommandManager class.

snowbanana12345 committed:
placed all classes under package duke

snowbanana12345 committed:
fix issue with packaging
dukeSetDone now throws BadInputException when no task number is given, the task number is not valid or exceeds the memory space provided to duke

snowbanana12345 committed:
add catch NumberFormatException to dukeSetDone Method to handle not user not providing a number after done command

snowbanana12345 committed:
wrote the user guide

snowbanana12345 committed:
update user guide

snowbanana12345 committed:
rewrite the whole code in OOP style

snowbanana12345 committed:
caught up to week 4 work

snowbanana12345 committed:
caught up to week 5 work

snowbanana12345 committed:
implement Delete functionality

snowbanana12345 committed:
fix bug that case COMMAND_DELETE did not have a break
updated the user guide to reflect the new delete functionality

manaswini05 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,317 @@
 public class Duke {
+    static Task[] storedUserTasks = new Task[100];
+    static int numberStoredTasks = 0;
+    static boolean dukeActive = false;
```
try to follow the naming convention for boolean - isDukeActive

manaswini05 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,317 @@
+     * prints a greeting to the user
+     */
+    public static void dukeGreet(){
+        printHorizontalLine(50);
```
50 looks like a magic number - maybe can declare as a constant

manaswini05 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,317 @@
+            throw new EmptyInputException("The description of a Event cannot be empty!");
+        }
+        StringBuilder taskName = new StringBuilder();
+        StringBuilder at = new StringBuilder();
```
at is a vague name- can consider changing the name

manaswini05 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,317 @@
+
+    private static void dukeAddToDo(String[] userInputs)
+            throws EmptyInputException{
+        ToDo new_task = createToDo(userInputs);
```
consider following the naming convention for new_task

manaswini05 reviewed (commented):
Good! It would be great if you followed the same pattern for your comments.

snowbanana12345 committed:
implement save

snowbanana12345 committed:
Merge branch 'branch-level-7'

# Conflicts:
#	docs/README.md
#	src/main/java/duke/DukeManager.java

snowbanana12345 committed:
fix messy merge conflicts

snowbanana12345 committed:
clean up

snowbanana12345 committed:
remove dead code

snowbanana12345 committed:
add load functionality

snowbanana12345 committed:
add support for duke to recognise date time
has not implemented fully

snowbanana12345 committed:
add function isOnDate to Event class
it returns true if it contains a valid date and has taken in a valid date and the dates are equal
returns false otherwise.

snowbanana12345 committed:
did the same for deadline

snowbanana12345 committed:
implement list-by-date /time YYYY-MM-DD-TTTT command for user.
user can see the list of all tasks with the specified date time
the task number is still incorrect

snowbanana12345 committed:
implement find functionality

snowbanana12345 committed:
create class input parser and extracted method in DukeManager into that class
fixed problem with input parsing removing all spaces

snowbanana12345 committed:
resolve merge conflicts

snowbanana12345 committed:
minor bug fixes

snowbanana12345 committed:
Merge branch 'master' into heads/branch-level-9

snowbanana12345 committed:
Merge pull request #2 from snowbanana12345/heads/branch-level-9

implement find functionality

snowbanana12345 committed:
debugged

snowbanana12345 committed:
restore un pushed code

snowbanana12345 committed:
fixxed some unpushed features

snowbanana12345 committed:
hahaha

snowbanana12345 committed:
fixed a bug where isOnDate method also checks if the time is equal.

snowbanana12345 committed:
Set theme jekyll-theme-merlot

snowbanana12345 committed:
improve code quality

snowbanana12345 committed:
add java doc

snowbanana12345 committed:
java doc done

snowbanana12345 committed:
reset done and not done symbol

snowbanana12345 committed:
Merge branch 'master' of https://github.com/snowbanana12345/ip

snowbanana12345 committed:
updated user guide
